From 3052b073afc10992acd0c84f005be50c1323f4f8 Mon Sep 17 00:00:00 2001
From: Etienne Carriere <etienne.carriere@foss.st.com>
Date: Thu, 13 Jun 2024 16:48:38 +0200
Subject: [PATCH 1119/1695] core: add mutex_is_locked() helper function

Add mutex_is_locked() helper function to return whether a mutex is
locked or not. This helper function must be use with care since it
does not guarantee that the mutex is held by the executing thread.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 93cf2682a5fa880b2fc6d82715e99bd7062284f6 core: add mutex_is_locked() helper function]
Change-Id: Ica26e667467c5016eebaa2c5b9f96fae4071d399
[cherry-picked from commit 30730d67b82f150d41be74118851cbb646b8a3ec]
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/393770
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
---
 core/include/kernel/mutex.h | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/core/include/kernel/mutex.h b/core/include/kernel/mutex.h
index c701188d7..a102eae3f 100644
--- a/core/include/kernel/mutex.h
+++ b/core/include/kernel/mutex.h
@@ -101,5 +101,14 @@ void condvar_broadcast(struct condvar *cv);
 void condvar_wait(struct condvar *cv, struct mutex *m);
 #endif
 
+/*
+ * Helper for testing that a given mutex is locked. This helper is to be used
+ * with caution since it does not enforce that the executing thread is holding
+ * the mutex.
+ */
+static inline bool mutex_is_locked(struct mutex *m)
+{
+	return m->state;
+}
 #endif /*__KERNEL_MUTEX_H*/
 
-- 
2.47.2

